scripts/zones/Western_Altepa_Desert/Zone.lua

```diff
@@ -96,15 +103,15 @@ zoneObject.onZoneWeatherChange = function(weather)
         -- 10% chance for KV pop at start of single earth weather
         local chance = math.random(1, 10)
         if chance == 1 then
-            DisallowRespawn(kingV:getID(), false)
+            DisallowRespawn(ID.mob.KING_VINEGARROON, false)
```

Comment:
Having this in onZoneTick will spawn him 100% of the time as it will continue to call this every 4000 ms I believe. It should stay in onWeatherChange

Reply:
I might have to add variable to make sure only checked once in single weather. It cannot be in on weather change because we want it to spawn if the window opens during double or single earth

Reply:
You can set a variable to control the attempt of spawning within onWeatherChange. If this if statement fires while this variable is set, reset it to 0. Since this variable is only ever resetting once per weather change, it will ensure only 1 10% check is done.

Reply:
exactly like that! lol

Reply:
> exactly like that! lol

Yep, seems to work well now, at least for all scenarios I tested, thanks!